### PR TITLE
Add AOSC support

### LINKED ISSUE #292: Add AOSC support
List of packages: https://packages.aosc.io/repo/amd64

### PR DESCRIPTION
**Pull request by gumblex**
Fixes #292. Didn't test, should work. @l2dy

### CODE REVIEW

**Comment by l2dy** [comment]
@AMDmi3 Have you reviewed this pull request?

File: repology/parsers/aosc.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Copyright (C) 2017 Dmitry Marakasov <amdmi3@amdmi3.ru>
```
Comment (AMDmi3) [inline repology/parsers/aosc.py]: I'm not the author :)

File: repology/parsers/aosc.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# repology is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with repology.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import os
```
Comment (AMDmi3) [inline repology/parsers/aosc.py]: Not needed

File: repology/parsers/aosc.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import re
```
Comment (AMDmi3) [inline repology/parsers/aosc.py]: Not needed

File: repos.d/aosc.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+###########################################################################
+# AOSC OS
+###########################################################################
+- name: aosc
+  type: repository
+  desc: AOSC OS Packages
```
Comment (AMDmi3) [inline repos.d/aosc.yaml]: Is "OS" part really necessary? I don't see it widely used on e.g. https://packages.aosc.io/, and it adds a bit of visual noise, while providing no additional information.

**Comment by AMDmi3** [comment]
There's also a problem with package quality: AOSC does not mark python module packages with suffix like it does with `perl-` or `ruby-`. Creating rules for all python packages is not an option.

**Comment by l2dy** [comment]
`pkg.category` could be used to add prefix to all python packages.

**Comment by AMDmi3** [comment]
I'm afraid it'd add prefix to ports that don't need it. For instance, python itself.

**Comment by l2dy** [comment]
In that case we can create rules for them.

**Comment by gumblex** [comment]
So what's Gentoo's solution?

**Comment by AMDmi3** [comment]
Gentoo's `dev-python` category does seem to only contain python modules, so adding prefix based on category works there.

**Comment by l2dy** [comment]
@AMDmi3: What's your opinion on adding prefix based on category and create rules for exceptions?
As a last resort, we can ignore all python packages.

@gumblex: @AMDmi3 asked you to squash your commits in the review summary, did you miss it?

**Comment by gumblex** [comment]
@l2dy How to do that squashing?

**Comment by l2dy** [comment]
@gumblex `git rebase -i @~3` (3 commits in your PR), and replace `pick` with `fixup` in the last two commits.

**Comment by l2dy** [comment]
@gumblex I can squash the commits for you if you grant me push access to https://github.com/AOSC-Dev/repology.

**Comment by AMDmi3** [comment]
Just leave it be :) I'll try to merge it tomorrow.

**Comment by l2dy** [comment]
@AMDmi3 Any progress?

**Comment by AMDmi3** [comment]
Merged, thank you!

I've had to add a bunch of rules to fix package names and versions, some of these should be fixed upstream IMO, including:

- `jade` -> `openjade`
- `pygeoip` -> `geoip` (both modules exist, `pygeoip` `0.3.2` and `geoip` `1.3.2`; AOSC has `pygeoip 1.3.2` which is actually `geoip`)
- `mpc` -> `libmpc` 
- missing `rc1` suffix in `ibus-sunpinyin` and `sunpinyin`
- a bunch of snapshot versions (`grep aosc rules.d/20-verignores.yaml`), see #345 on my idea how to fix these

Also I've had to ignore a bunch of `999` versions globally and add some code to handle `systemd-cron-next 1:1.0.2`, I assume that is an epoch and more versions like that may appear

File: repos.d/aosc.yaml (hunk 4 above)
Comment (MingcongBai) [reply repos.d/aosc.yaml]: @AMDmi3 Yes, I do apologise if our website is not explicit enough about this information. "AOSC" refers to our community name "Anthon Open Source Community", while "AOSC OS" refers to our actual Linux distribution project.

**Comment by AMDmi3** [comment]
@gumblex https://packages.aosc.io/list.json gives error 500

**Comment by l2dy** [comment]
@gumblex fixed the issue.

**Pull request closed without merging**